core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/newui/BinaryParsTab.java

```diff
@@ -224,42 +231,38 @@ public void setVisible(boolean _visible) {
 
 	@Override
 	public void updateData(ICResourceDescription cfgd) {
-		String[] ids = null;
+		Set<String> selectedIds = new LinkedHashSet<>();
 		if (page.isForPrefs()) { // prefs
 			if (cfgd != null && cfgd.getConfiguration() != null) {
 				tps = cfgd.getConfiguration().getTargetPlatformSetting();
-				if (tps != null)
-					ids = tps.getBinaryParserIds();
+				if (tps != null) {
+					String[] binaryParserIds = tps.getBinaryParserIds();
+					if (binaryParserIds != null) {
+						selectedIds.addAll(Arrays.asList(binaryParserIds));
+					}
+				}
 			}
-			if (ids == null)
-				ids = new String[0]; // no selection
 		} else { // project
 			ICConfigurationDescription[] cfgs = page.getCfgsEditable();
-			ids = CoreModelUtil.getBinaryParserIds(cfgs);
-		}
-		Object[] data = new Object[configMap.size()];
```

Comment:
The key error is that size of data was wrong, it assumed that it needed to be the size of all plugin.xml defined binary parsers. It didn't account for ids having other ids, so the i variable below would be over incremented.

All this is solved by removing hard coded array sizes.